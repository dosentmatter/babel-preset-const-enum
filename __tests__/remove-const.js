import { transformAsync } from '@babel/core';
import preset from '../src';

const options = {
  filename: 'file.ts',
  presets: [preset],
};

it('Transforms no initializers', async () => {
  const input = `const enum Direction { Left, Right, Down, Up };
`;

  const { code: output } = await transformAsync(input, options);
  expect(output).toMatchSnapshot();
});

it('Transforms string members', async () => {
  const input = `const enum Enum {
  A = 1,
  B = A,
  C = '',
  D = C,
  E = 1,
  F,
}
`;

  const { code: output } = await transformAsync(input, options);
  expect(output).toMatchSnapshot();
});

it('Transforms computed members', async () => {
  const input = `const enum MyEnum {
  A = 1,
  B = A,
  C,
  D = C,
  E = 1,
  F,
  G = A * E,
  H = A ** B ** C,
  I = A << 20
}
`;

  const { code: output } = await transformAsync(input, options);
  expect(output).toMatchSnapshot();
});
