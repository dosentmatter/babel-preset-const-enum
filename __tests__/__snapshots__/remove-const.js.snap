// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transforms computed members 1`] = `
"\\"use strict\\";

enum MyEnum {
  A = 1,
  B = A,
  C,
  D = C,
  E = 1,
  F,
  G = A * E,
  H = A ** B ** C,
  I = A << 20,
}"
`;

exports[`Transforms no initializers 1`] = `
"\\"use strict\\";

enum Direction {
  Left,
  Right,
  Down,
  Up,
}
;"
`;

exports[`Transforms string members 1`] = `
"\\"use strict\\";

enum Enum {
  A = 1,
  B = A,
  C = '',
  D = C,
  E = 1,
  F,
}"
`;
